storage/txcache/txCacheFailsafe.go

```diff
@@ -0,0 +1,140 @@
+package txcache
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+var _ storage.Cacher = (*txCacheFailsafeDecorator)(nil)
+var _ txCache = (*txCacheFailsafeDecorator)(nil)
+
+// txCacheFailsafeDecorator is a decorator over a cache, a decorator which wraps some operations in try / catch constructs
+type txCacheFailsafeDecorator struct {
+	backingCache txCache
```

Comment:
If this structure is really needed, could have used direct embedding and overload of the required methods. A lot of this code is wrapped code. Please take a look at p2p/libp2p/discover/hostDecorator.go

Reply:
Removed, see above. Indeed, `hostDecorator` is a good example.